### DISCUSSION TITLE
Compile Required Delegates Source with IM

### DISCUSSION THREAD

**Discussion by paulasiimwe** [ImageMagick/ImageMagick#2531] [Help] [answered]
Hey all,

Is there a way to compile IM with Delegates from source by grabbing all the needed Delegate's source folders, and including them in the IM source folder so that they are compiled first and then IM's own?

Grabbing the delegates source folders and dumping them into the IM folder, png, webp, jpeg, heif etc is the easy part, I imagine I must make adjustments to IM's configure or makefile? Which adjustments are these?  Any examples would be helpful.

**Comment by snibgo** [comment]
What is your platform?

What you ask for is pretty much how the Windows source installation works. However, there isn't really a definition of "needed" delegates. For example, I have:
```
f:\web\im>%IM7DEV%magick -version

Version: ImageMagick 7.0.8-64 Q32 x86_64 2019-12-18 https://imagemagick.org
Copyright: © 1999-2019 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(4.5)
Delegates (built-in): bzlib cairo fftw fontconfig fpx freetype jbig jng jpeg
 lcms ltdl lzma pangocairo png rsvg tiff webp wmf x xml zlib
```

**Reply by fmw42** [reply]
On Mac, you can install delegates via MacPorts and then install ImageMagick from source. I do that all the time. See https://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=21502&p=88202&hilit=OSX+MacPorts#p88202 and https://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=29100

**Comment by paulasiimwe** [comment]
Thanks @snibgo @fmw42  for your answers. 
My plan is to have a portable ready-to-compile IM with my required delegate's jpeg,png,tiff,heif,webp,raw,gs,rsvg source files as well in the IM folder... zip it and use across multiple platforms, windows, mac or linux, as I'll have to do this multiple times. 

I'd rather have it done this way than first go through the ritual of each platform's delegate installation be that sudo-apt get for linux or macports for Mac. 

I know the windows source releases [here](https://imagemagick.org/download/windows/releases/) normally come with as many delegates as possible in the unarchived folder, but it seems one still has to compile each delegate one by one first

**Comment by fmw42** [comment, marked as answer]
The Unix binary listed as magick at https://imagemagick.org/script/download.php is a portable ready to use binary. It might work also for Mac OSX.
